trlx/pipeline/ppo_pipeline.py

```diff
@@ -93,31 +35,34 @@ def create_loader(
         self,
         batch_size: int,
         shuffle: bool,
-        prep_fn: Callable = None,
-        num_workers: int = 0,
     ) -> DataLoader:
-        """
-        Create dataloader for the sentiment task.
-
-        :param prep_fn: Should be tokenizing function
-        :type prep_fn: Callable[Iterable[str], Dict[str, torch.tensor]]
-        """
-        # TODO(dahoas): Decide how to support varying sizes of prompts without having to tokenize on fly
         def collate_fn(elems: Iterable[PPORLElement]):
-            res = PPORLBatch(
-                torch.stack(
-                    [elem.query_tensor for elem in elems]
-                ),  # Assumes token tensors all same size
-                torch.stack([elem.response_tensor for elem in elems]),
-                torch.stack([elem.logprobs for elem in elems]),
-                torch.stack([elem.values for elem in elems]),
-                torch.stack([elem.rewards for elem in elems]),
+            return PPORLBatch(
+                # Left padding of already left-padded queries
+                pad_sequence(
+                    [elem.query_tensor.flip(0) for elem in elems],
+                    padding_value=self.pad_token_id,
+                    batch_first=True,
+                ).flip(1),
+                # Right pad the rest, to have a single horizontal query/response split
+                pad_sequence(
```

Comment:
Ok so seeing this now I assume the padded values are handled in loss computation by the attention mask?